check_license_source_files.sh

```diff
@@ -0,0 +1,202 @@
+    local -r file="${1}"
+    awk '!(NR == 1 && $1 ~/^#!.*/)' "${file}" > "${tf}"
+    echo "${tf}"
+}
+
+check_file() {
+    local file="$1"
+
+    local license
+    local lines
+    local lic_type
+    local -r CM="$2"
+
+    cat > license-os.tmp <<-EOF
+${CM}    Licensed under the Apache License, Version 2.0 (the "License");
+${CM}    you may not use this file except in compliance with the License.
+${CM}    You may obtain a copy of the License at
+${CM}
+${CM}        http://www.apache.org/licenses/LICENSE-2.0
+${CM}
+${CM}    Unless required by applicable law or agreed to in writing, software
+${CM}    distributed under the License is distributed on an "AS IS" BASIS,
+${CM}    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+${CM}    See the License for the specific language governing permissions and
+${CM}    limitations under the License.
+EOF
+    LINES_OS=$(cat license-os.tmp | wc -l)
+    # we need to add two extra lines missing from the license preamble
+    # // Copyright <copyright_year> Northern.tech AS
+    # //
+    LINES_OS=$(($LINES_OS + 2))
+
+    cat > license-ent.tmp <<-EOF
+${CM}    All Rights Reserved
+EOF
+    LINES_ENT=$(cat license-ent.tmp | wc -l)
+    # we need to add two extra lines missing from the license preamble
+    # // Copyright <copyright_year> Northern.tech AS
+    # //
+    LINES_ENT=$(($LINES_ENT + 2))
+
+    if is_enterprise ${file}; then
+        license="license-ent.tmp"
+        lines=$LINES_ENT
+        lic_type="Enterprise"
+    else
+        license="license-os.tmp"
+        lines=$LINES_OS
+        lic_type="Open Source"
+    fi
+
+    modified_year=$(git log --follow -n1 --format=%ad --date=format:%Y -- "$file")
+
+    orig_file="${file}"
+    file=$(strip_hashbang "${file}")
+
+    head -n $lines "$file" | tail -n +3 | diff -u "$license" - > /dev/null
+    if [ $? -ne 0 ]; then
+        echo >&2 "!!! FAILED license check on ${orig_file}. Expected this $lic_type license:"
+        cat "$license" >&2
+        TEST_RESULT=1
+    else
+        copyright_modified=$(echo "${CM} Copyright <copyright_year> Northern.tech AS" | sed "s/<copyright_year>/$modified_year/g")
+        copyright_file="$(head -n 1 "$file")"
+        if [ "$copyright_modified" != "$copyright_file" ]; then
+            echo >&2 "!!! FAILED license check on ${orig_file}; make sure copyright year matches last modified year of the file ($modified_year)"
+            TEST_RESULT=1
+        fi
+    fi
+    return ${TEST_RESULT}
+}
+
+check_files() {
+    if [[ ! $# -ge 1 ]]; then
+        echo >&2 "check_files requires one or more parameters"
+    fi
+    local -r SOURCE_FILES="$@"
+    for source_file in ${SOURCE_FILES}; do
+        case ${source_file} in
+          *.go)
+              CM='//'
+              ;;
+          *.py)
+              CM="#"
+              ;;
+          *)
+              echo >&2 "No source file testing done for file:\n\t ${source_file}\nThe filetype is unsupported type"
```

Comment:
This just continues to `check_file` below with `CM` unset. I would suggest setting `TEST_RESULT=1` and calling `continue` here, so that it loops around and we catch it.